roles/network_plugin/calico/tasks/pre.yml

```diff
@@ -1,20 +1,4 @@
 ---
-- name: Slurp CNI config
-  slurp:
-    src: /etc/cni/net.d/10-calico.conflist
-  register: calico_cni_config_slurp
-  failed_when: false
-
-- block:
-  - name: Set fact calico_cni_config from slurped CNI config
-    set_fact:
-      calico_cni_config: "{{ calico_cni_config_slurp['content'] | b64decode | from_json }}"
-  - name: Set fact calico_datastore to etcd if needed
-    set_fact:
-      calico_datastore: etcd
-    when: "'etcd_endpoints' in calico_cni_config.plugins.0"
-  when: calico_cni_config_slurp.content is defined
-
```

Comment:
The above lines have been added since https://github.com/kubernetes-sigs/kubespray/pull/6693
It is nice to get feedback from @hafe

Reply:
Those tasks read out from an already configured system what data store calico is using. I added that to protect existing systems during upgrade. Required because the default data store was changed